packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts

```diff
@@ -0,0 +1,252 @@
+import * as cdk from '@aws-cdk/core';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as iam from '@aws-cdk/aws-iam';
+import * as path from 'path';
+import { prepareApp } from '@aws-cdk/core/lib/private/prepare-app';
+import { AmplifyRootStackResourceProps, AmplifyRootStackTemplateProps } from './types';
+import { JSONUtilities, pathManager } from 'amplify-cli-core';
+import { Template } from 'cloudform-types';
```

Comment:
Why do we need `Template` type?

Reply:
I want a type similar to template as we are using cfn modifiers in our cli which required that. I can make my own Template type too

Reply:
We should do all the processing/modification using this class rather than passing a cfn JSON object as the CDK class can provide lot more utility like tree walking

Reply:
I am doing all the processing inside AmplifyRootTransform class which has this lifecycle

getInput() -> generateRootStackResource() -> applyoverride -> synthesizeTemplate -> deploytoDisk()

this all are called under transform() function , which returns template type which is then used in the CFNmodifiers like(IAM permission boundary and S3BucketEncryption).